python_modules/dagster/dagster/_config/pythonic_config/config.py

```diff
@@ -163,6 +163,9 @@ def __init__(self, **config_dict) -> None:
         modified_data = {}
         for key, value in config_dict.items():
             field = self.__fields__.get(key)
+            if field and not field.required and value is None:
+                continue
+
```

Comment:
Please comment (in code) as to why the v2 required this change and why it is backward compatible with v1